includes/reader-activation/class-reader-activation.php

```diff
@@ -2694,6 +2703,70 @@ public static function login_after_password_reset( $user ) {
 		self::set_current_reader( $user );
 	}
 
+	/**
+	 * Check if a URL matches an OAuth redirect route.
+	 *
+	 * @param string $url The URL to check.
+	 * @return bool True if the URL path contains an OAuth route.
+	 */
+	private static function is_oauth_redirect( $url ) {
+		$url_path = \wp_parse_url( $url, PHP_URL_PATH ) ?? '';
+
+		/**
+		 * Filters the list of OAuth routes that should redirect to RAS login.
+		 *
+		 * @param array  $routes Array of OAuth route paths.
+		 * @param string $url_path The URL path being checked.
+		 */
+		$routes = apply_filters( 'newspack_ras_oauth_redirect_routes', self::OAUTH_REDIRECT_ROUTES, $url_path );
+
+		return ! empty(
+			array_filter(
+				$routes,
+				fn( $route ) => str_contains( $url_path, $route )
+			)
+		);
+	}
+
+	/**
+	 * Redirect OAuth authorization requests to RAS login instead of wp-login.php.
+	 *
+	 * This enables OAuth clients to use the RAS login screen which provides
+	 * Google SSO and a better user experience while maintaining PKCE OAuth 2.0 compatibility.
+	 *
+	 * This is a generic OAuth + RAS integration, not specific to any particular OAuth client.
```

Comment:
I love that you made this general purpose so we can use it with other providers! Good idea!